client/signup/steps/woocommerce-install/confirm/index.tsx

```diff
@@ -16,16 +16,26 @@ import {
 	getEligibility,
 	EligibilityData,
 } from 'calypso/state/automated-transfer/selectors';
+import { submitSignupStep } from 'calypso/state/signup/progress/actions';
 import { getSiteDomain } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import type { WooCommerceInstallProps } from '../';
 
 import './style.scss';
 
 export default function Confirm( props: WooCommerceInstallProps ): ReactElement | null {
-	const { siteId, goToStep, isReskinned, stepSectionName, headerTitle, headerDescription } = props;
+	const {
+		goToStep,
+		isReskinned,
+		stepSectionName,
+		headerTitle,
+		headerDescription /*, siteId*/,
```

Comment:
I don't know how much I trust siteId coming from props, to me it looks like it could get changed in the controller async (.then()...). I might be wrong on this, couldn't quite figure it out. If I'm right the transfer step is risky until we know the site Id is certain and not going to change.

Reply:
The reason we're fetching the site again is in case the site slug changed due to a domain being purchased. So I suspect if any prop changed it'd be the `siteSlug`, not `siteId`

Reply:
Figured out what I was misunderstanding here, was non issue, its not calling next until the selected site is dispatched again.

Also site id isn't passed through props directly but `queryObject` or `signupDependencies`.